plum/function.py

```diff
@@ -125,21 +259,6 @@ def __doc__(self) -> Optional[str]:
 
         Upon instantiation, this property is available through `obj.__doc__`.
         """
-        try:
-            self._resolve_pending_registrations()
-        except NameError:  # pragma: specific no cover 3.7 3.8 3.9
-            # When `staticmethod` is combined with
-            # `from __future__ import annotations`, in Python 3.10 and higher
-            # `staticmethod` will attempt to inherit `__doc__` (see
-            # https://docs.python.org/3/library/functions.html#staticmethod). Since
-            # we are still in class construction, forward references are not yet
-            # defined, so attempting to resolve all pending methods might fail with a
-            # `NameError`. This is fine, because later calling `__doc__` on the
-            # `staticmethod` will again call this `__doc__`, at which point all methods
-            # will resolve properly. For now, we just ignore the error and undo the
-            # partially completed :meth:`Function._resolve_pending_registrations` by
-            # clearing the cache.
-            self.clear_cache(reregister=False)
```

Comment:
Could you explain why you've removed this? Do you think it's not at all necessary to resolve pending registration when getting the owner? I should've documentation why that line was there in the first place... I think removing it might make sense, but I'm worried about subtly breaking something that's currently not tested for, so I just want to double check with you.